android/java/org/chromium/chrome/browser/upgrade/BraveUpgradeJobIntentService.java

```diff
@@ -53,12 +54,51 @@
     private static final String DSE_NAME = "Google";
     private static final String DSE_KEYWORD = "google.com";
 
+    // Old tabs bottom toolbar settings
+    private static final String BOTTOM_TOOLBAR_ENABLED_KEY = "bottom_toolbar_enabled";
+
+    // To detect update from tabs
+    private static final String PREF_STATS_PREFERENCES_NAME = "StatsPreferences";
+    private static final String PREF_WEEK_OF_INSTALLATION_NAME = "WeekOfInstallation";
+
     public static void startMigrationIfNecessary(Context context) {
+        if (BraveUpgradeJobIntentService.needToMigratePreferences()) {
+            // Migrate bottom toolbar settings
+            SharedPreferences sharedPreferences = ContextUtils.getAppSharedPreferences();
+            SharedPreferences.Editor sharedPreferencesEditor = sharedPreferences.edit();
+            sharedPreferencesEditor.putBoolean(BravePreferenceKeys.BRAVE_BOTTOM_TOOLBAR_ENABLED_KEY,
+                sharedPreferences.getBoolean(BOTTOM_TOOLBAR_ENABLED_KEY, true));
+            sharedPreferencesEditor.apply();
+        }
         // Start migration in any case as we can have only partial data
         // to migrate available
         BraveUpgradeJobIntentService.enqueueWork(context, new Intent());
     }
 
+    private static boolean needToMigratePreferences() {
+        SharedPreferences sharedPreferences = ContextUtils.getAppSharedPreferences();
+        boolean migrated = sharedPreferences.getBoolean(BraveHelper.PREF_TABS_SETTINGS_MIGRATED, false);
+        if (migrated) {
+            // Everything was already migrated
+            return false;
+        }
+
+        // Detect whether it is update from tabs
+        SharedPreferences.Editor sharedPreferencesEditor = sharedPreferences.edit();
+        SharedPreferences prefStatsFromTabs = ContextUtils.getApplicationContext()
+                .getSharedPreferences(PREF_STATS_PREFERENCES_NAME, 0);
+        boolean updateFormTabs = prefStatsFromTabs.contains(PREF_WEEK_OF_INSTALLATION_NAME);
+        if (!updateFormTabs) {
```

Comment:
We are confident that it's a tabs=>core update by checking the preferences above